desc/objectives/_neoclassical.py

```diff
@@ -0,0 +1,263 @@
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute Jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    rho : ndarray
+        Unique coordinate values specifying flux surfaces to compute on.
+    alpha : ndarray
+        Unique coordinate values specifying field line labels to compute on.
+    knots_per_transit : int
+        Number of points per toroidal transit at which to sample data along field
+        line. Default is 100.
+    num_transit : int
+        Number of toroidal transits to follow field line.
+        For axisymmetric devices, one poloidal transit is sufficient. Otherwise,
+        more transits will give more accurate result, with diminishing returns.
+    num_quad : int
+        Resolution for quadrature of bounce integrals. Default is 32.
+    num_pitch : int
+        Resolution for quadrature over velocity coordinate. Default 50.
+    batch : bool
+        Whether to vectorize part of the computation. Default is true.
+    num_well : int
+        Maximum number of wells to detect for each pitch and field line.
+        Default is to detect all wells, but due to limitations in JAX this option
+        may consume more memory. Specifying a number that tightly upper bounds
+        the number of wells will increase performance.
+    name : str, optional
+        Name of the objective function.
+    jac_chunk_size : int , optional
+        Will calculate the Jacobian for this objective ``jac_chunk_size``
+        columns at a time, instead of all at once. The memory usage of the
+        Jacobian calculation is roughly ``memory usage = m0 + m1*jac_chunk_size``:
+        the smaller the chunk size, the less memory the Jacobian calculation
+        will require (with some baseline memory usage). The time to compute the
+        Jacobian is roughly ``t=t0 +t1/jac_chunk_size``, so the larger the
+        ``jac_chunk_size``, the faster the calculation takes, at the cost of
+        requiring more memory. A ``jac_chunk_size`` of 1 corresponds to the least
+        memory intensive, but slowest method of calculating the Jacobian.
+        If None, it will use the largest size i.e ``obj.dim_x``.
+
+    """
+
+    _coordinates = "r"
+    _units = "~"
+    _print_value_fmt = "Effective ripple ε: "
+
+    def __init__(
+        self,
+        eq,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        rho=np.linspace(0.5, 1, 3),
```

Comment:
why 3 surfaces? I feel like it would be better to default just to rho=1 which is what we do for some of the QS objectives, or an actual grid over radius with rho ~ L_grid

Reply:
AD consumed more memory then we have for the optimizations I ran when if there are many surfaces. Maybe this will change with the new batched stuff. I can change to rho=1.